### PR TITLE
Add `UkAccountNumber` and `UkSortCode` asserts

### PR DESCRIPTION
**Pull request by promag**
R=@ruimarinho

### CODE REVIEW

**Comment by ruimarinho** [comment]
VOCALINK provides a list of sort codes for validation already (`scsubtab.txt`). This regex only appears to be validating the format instead.

We should - potentially in addition to what's currently being tested - test the sort code range against that file:

```
input >= start && input <= end
```

**Comment by ruimarinho** [comment]
We will also need to implement Modulus Checking.

**Comment by ruimarinho** [comment]
Here is a possible implementation for Modulus Checking - https://github.com/liamja/modcheck.js (also implicitly includes sort code validation).

File: README.md [hunk 1]
```diff
@@ -162,6 +163,9 @@ Tests if the phone is valid and optionally if it belongs to the given country. T
 ### PlainObject
 Tests if the value is a plain object.
 
+### UkSortCode
+Tests if the value is a valid UK Sort Code.
```
Comment (promag) [inline README.md]: Missing update.

File: src/asserts/voca-link-account-number-assert.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  const Modcheck = require('modcheck');
+
+  /**
+   * Class name.
+   */
+
+  this.__class__ = 'VocaLinkAccountNumber';
+
+  /**
+   * Check if `accountNumber` is defined.
```
Comment (franciscocardoso) [inline src/asserts/voca-link-account-number-assert.js]: sortCode?

File: src/asserts/voca-link-account-number-assert.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * Module dependencies.
+ */
+
+import { Validator, Violation } from 'validator.js';
+
+/**
+ * Export `VocaLinkAccountNumberAssert`.
+ */
+
+export default function vocaLinkAccountNumberAssert(sortCode) {
```
Comment (franciscocardoso) [inline src/asserts/voca-link-account-number-assert.js]: `vocaLinkAccountNumberAssert` and you using `sortCode`. Is this ok?

Comment (franciscocardoso) [inline src/asserts/voca-link-account-number-assert.js]: Hum, you need both I guess :)

File: src/asserts/voca-link-account-number-assert.js (hunk 2 above)
Comment (promag) [inline src/asserts/voca-link-account-number-assert.js]: :+1:

File: README.md [hunk 4]
```diff
@@ -162,6 +164,18 @@ Tests if the phone is valid and optionally if it belongs to the given country. T
...
 Tests if the value is a plain object.
 
+### UkAccountNumber
+Tests if the value is a valid UK account number.
+
+#### Arguments
+- `sortCode` (optional) - the corresponding sort code for [modulus checking](https://www.vocalink.com/customer-support/modulus-checking/).
+
+### UkSortCode
+Tests if the value is a valid UK sort code.
```
Comment (ruimarinho) [inline README.md]: Link to https://en.wikipedia.org/wiki/Sort_code?

**Comment by ruimarinho** [comment]
Current PR LGTM. We just need to submit our patches to modcheck.js upstream and as soon as a new version is release, merge this too.

**Comment by franciscocardoso** [comment]
Please hold the merge of this please. The module used to validate the uk-sorting-codes has a few issues. We are working in a better one. I'll let you know as son as we have that covered.

/cc @ruimarinho

**Comment by franciscocardoso** [comment]
Deprecated by https://github.com/seegno/validator.js-asserts/pull/79.

**Pull request closed without merging**